### PR TITLE
KAYAK-866 Extract @Kazark's abstractions from monorepo

### PR DESCRIPTION
**Pull request by amohrland**
[KAYAK-866] Extract @Kazark's abstractions from monorepo

[KAYAK-866]: https://banno-jha.atlassian.net/browse/KAYAK-866

### CODE REVIEW

File: core/src/main/scala/com/banno/kafka/TopicPartitionOffset.scala [hunk 1]
```diff
@@ -1,26 +0,0 @@
-/*
```
Comment (Kazark) [inline core/src/main/scala/com/banno/kafka/TopicPartitionOffset.scala:1]: I think I'm missing context. Why did this one get dropped?

**Review by Kazark — APPROVED** [review]

File: core/src/main/scala/com/banno/kafka/TopicPartitionOffset.scala (hunk 1 above)
Comment (amohrland) [reply core/src/main/scala/com/banno/kafka/TopicPartitionOffset.scala:1]: It's subsumed by https://github.com/Banno/kafka4s/pull/425/files#diff-6d08a03f2b6664bedea88f00caebbc2f2ace923dc3371a5e8dba0fb9b425e89bR11. It's a breaking change, but we're on milestones, and we make no guarantees. For the time being, I turned the implicit `toMap` conversion into an explicit method.

**Review by Kazark — COMMENTED** [review]
Do we need to add copyright headers to these files?

File: core/src/main/scala/com/banno/kafka/TopicPartitionOffset.scala (hunk 1 above)
Comment (Kazark) [reply core/src/main/scala/com/banno/kafka/TopicPartitionOffset.scala:1]: Got it. I had a vague memory of something like that but couldn't remember what file it was in.

**Comment by amohrland** [comment]
> copyright headers

Good call 👍. Looks like we were conditionally checking headers for jdk 8, probably to avoid duplicate checks. With that conditional removed, the check failed as expected. I'll get them added.

**Comment by amohrland** [comment]
Headers added; there were some test files and examples that also needed them. I'll forward-port to `master`.

**Pull request merged**